frontend_tests/casper_tests/16-copy-and-paste.js

```diff
@@ -140,10 +140,10 @@ casper.then(function () {
     var actual_copied_lines = copy_messages('copy paste test A', 'copy paste test C');
     var expected_copied_lines = [
         'Verona > copy-paste-subject #1 Today',
-        'Iago: copy paste test A',
-        'Iago: copy paste test B',
+        'copy paste test A',
+        'copy paste test B',
         'Verona > copy-paste-subject #2 Today',
-        'Iago: copy paste test C',
+        'copy paste test C',
```

Comment:
Also, @rishig is it your intention that we suppress showing the sender on multiple messages even in cases where there are multiple topics involved?

Reply:
yeah, I think that's cleanest, and allows us to use a descriptoin like "copy selected text"? I don't feel strongly about it though; I think it's not going to be a common use of this feature.

Reply:
Note that the message sender will be visually selected, so I think a user would expect "copy selected text" to include the senders.  

For that reason and also for reasons of hierarchy of concept, I think it makes more sense to not suppress the sender when we have multiple recipient blocks.

Reply:
hm, on Chrome the sender, time, date line, and topic are not visually selected.
(Currently all 4 are copied though, in this PR)

Reply:
I guess my proposal is:
* Selection only selects text (which is the current behavior in chrome, for me. I'm on chrome 69).
* Copy selected text copies that text
* Copy messages includes sender, time, date changes, topic, etc.

Some reasons are
* "Copy selected text" is the only thing I've wanted in practice, and is also the only thing I've ever seen on czo. I can't think of a time I've seen someone quote the output of the copy-paste behavior we've had in place for the last 2+ years. 
* "Copy messages" makes sense for archival or if you're selecting large numbers of messages, and we should optimize it for that case.

Reply:
The current PR behavior does some things that I expect users to find broken (e.g. being unable to select a user's name to copy-paste just that).  We can try test-deploying to see to what extent people complain in our community, but I'm confident we'll get complaints.

@vinitS101 in any case, we should split out the changes to what "copy selected text" does (i.e. all the `no-select` / `auto-select` changes) into separate commit(s) from the new "copy selected messages" feature and menu so that we can potentially incrementally merge the new "Copy/paste" paste menu.  We'd want to do that separation regardless of what we ultimately decide here, because the code changes in this PR are pretty hard to understand with those two types of changes squashed together.

Reply:
hm, for some reason the selection behavior is behaving differently for me now than before.
Maybe worth talking about what we want here before test-deploying, since I think there are a number of different possibilities.

Reply:
I've broken it down to smaller commits and each "button"/feature is added in a separate commit. Behavior wise if there are any issues then please let me know.

Reply:
What about the following?
* Selection selects name and text. It does not select time, anything in the recipient bar, or date changes
* "Copy text" copies just the text
* "Copy messages" does a "full" copy
* "Quote and reply" I don't have strong feelings about
* Ctrl+C is not that important anymore, but I guess it could copy name and text?

Reply:
> Selection selects name and text. It does not select timestamp, anything in the recipient bar, or date changes

I missed the `no-select` for the dates shown in Group PM headers, so that might be something you saw.
Other than that, I think everything else in the message view, except the messages themselves, are not selectable.

I am not sure when you want the names to be select-able. Can you please elaborate on that? 

The current version of what's selected and what isn't is based off of our previous discussion [here](https://chat.zulip.org/#narrow/stream/101-design/topic/Menu.20for.20quote.26reply.20and.20copying.20when.20selecting.20text/near/721788).

Reply:
cool. The most recent proposal is for the name to be selected, if it's in the message feed (but not if it's in the recipient bar). Does that make sense?

Reply:
I can't say I fully understand this. Do you mean that sender names in the normal message view in streams (or PMs) should be selectable, whilst not being selectable in the the PM "headers/recipient bars"?

These names not being selectable was done so that users could select and copy message text and whilst omitting user names. The copying in this custom case simply copies all of the highlighted text to the clipboard.
If we were to make this change, then this copied text will contain the names of the users as well (if the names are also highlighted/selected).

As far as the "recipient bars" go, they are already set to `no-select` I believe.

Reply:
> Do you mean that sender names in the normal message view in streams (or PMs) should be selectable, whilst not being selectable in the the PM "headers/recipient bars"?

yup!

> If we were to make this change, then this copied text will contain the names of the users as well (if the names are also highlighted/selected).

yeah, so in this proposal all the buttons would do something custom, and only Ctrl+C would copy exactly the selected text. 

We should wait for sign-off from Tim on this idea (https://github.com/zulip/zulip/pull/11949#discussion_r285863378) though.

Reply:
Ah, okay, yes. That should work for most custom cases you mentioned in that comment.

Reply:
Yes, that all makes sense, with I guess the revision that names should still be selectable in "Group PMs" recipient bars, since those will not duplicate the actual senders.

Reply:
hm, if we're doing that I think it might be cleaner to just select stream and topic in all recipient bars. Thoughts?

Reply:
The concrete proposal on czo is [here](https://chat.zulip.org/#narrow/stream/101-design/topic/Menu.20for.20quote.26reply.20and.20copying.20when.20selecting.20text/near/750420)